### PR TITLE
Add tests for SurveyNode.

### PR DESCRIPTION
**Pull request by cassierecher**

### CODE REVIEW

File: claat/nodes/survey_test.go [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestNewSurveyNode(t *testing.T) {
+	tests := []struct {
+		name     string
+		inID     string
```
Comment (DragonRider0o0) [inline claat/nodes/survey_test.go:12]: Consider dropping the `in` prefix; `id` and `groups` are sufficient.

File: claat/nodes/survey_test.go [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+		out      *SurveyNode
+	}{
+		{
+			name: "Empty",
+			out: &SurveyNode{
+				node: node{typ: NodeSurvey},
+			},
+		},
+		{
+			// TODO: should the absence of an ID be an error?
```
Comment (DragonRider0o0) [inline claat/nodes/survey_test.go:23]: Good question, I think it should either evaluate as empty or as a valid depending on the desired behavior for SurveyNode.

File: claat/nodes/survey_test.go [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+				return
+			}
+		})
+	}
+}
+
+func TestSurveyNodeEmpty(t *testing.T) {
+	tests := []struct {
+		name string
+		in   *SurveyNode
```
Comment (DragonRider0o0) [inline claat/nodes/survey_test.go:128]: Use the same pattern for both tests, either pass the groups and ID in both or pass the SurveyNode in both.

Comment (cassierecher) [reply claat/nodes/survey_test.go:128]: Discussion on similar comment https://github.com/googlecodelabs/tools/pull/596/files/591b1ac68607383d635229cd063ed8561fc4b93d

File: claat/nodes/survey_test.go (hunk 1 above)
Comment (cassierecher) [reply claat/nodes/survey_test.go:12]: Discussion on similar comment https://github.com/googlecodelabs/tools/pull/594/files/8acfb33186a4e580c372a34d41087c721343dd3d#r666514627

File: claat/nodes/survey_test.go (hunk 2 above)
Comment (cassierecher) [reply claat/nodes/survey_test.go:23]: Yeah, we need to figure out what the desired behavior should be. I would like to defer this to a separate PR

Comment (DragonRider0o0) [reply claat/nodes/survey_test.go:23]: SGTM

**Review by shawnbuso — APPROVED** [review]

**Pull request merged**